odev/common/odoobin.py

```diff
@@ -710,6 +711,49 @@ def check_addons_path(cls, path: Path) -> bool:
         globs = (path.glob(f"*/__{manifest}__.py") for manifest in ["manifest", "openerp"])
         return path.is_dir() and any(manifest for glob in globs for manifest in glob)
 
+    @classmethod
+    def version_from_addons(cls, path: Path) -> Optional[OdooVersion]:
+        """Find the Odoo version of the addons in a directory, if this is an addons directory.
+
+        :param path: Path to the addons directory.
+        """
+        if not cls.check_addons_path(path):
+            return None
+
+        for addon in path.iterdir():
+            for manifest_name in ["__manifest__.py", "__openerp__.py"]:
+                manifest_path = addon / manifest_name
+
+                if not manifest_path.exists():
+                    continue
+
+                manifest = cls.read_manifest(manifest_path)
+
+                if manifest is None:
+                    continue
+
+                version = manifest.get("version", None)
+
+                if isinstance(version, str):
+                    return OdooVersion(version)
```

Comment:
Souldn't we take the highest value ? ( I mean , if one of the module didn't get upgraded yet but the others yes ? -> In progress upgrade )

Reply:
Yes this is a great idea!
Is topped at the fact that Odoo will crash at startup when some module are in an older version, but I'm not sure it does if the module is not installed

Reply:
Done!